bookmark-service/src/main/java/fr/loicmathieu/bookmarkit/BookmarkResource.java

```diff
@@ -20,38 +27,58 @@
 public class BookmarkResource {
 
     @GET
-    public List<Bookmark> listAll(){
+    @Operation(summary = "List all bookmarks")
+    @Counted(name = "listBookmarks.count")
+    @Timed(name = "listBookmarks.time")
+    public List<Bookmark> listBookmarks() {
         return Bookmark.listAll();
     }
 
     @GET
-    @Path("/{id}")
-    public Bookmark get(@PathParam("id") Long id) {
+    @Path("{id}")
+    @Operation(summary = "Get a bookmark")
+    @Counted(name = "getBookmark.count")
+    @Timed(name = "getBookmark.time")
+    public Bookmark getBookmark(@PathParam("id") Long id) {
         return Bookmark.findById(id);
     }
 
     @POST
     @Transactional
-    public Response create(Bookmark bookmark){
-        bookmark.persist();
-        return Response.created(URI.create("/bookmarks/" + bookmark.id)).build();
+    @Operation(summary = "Create a bookmark")
+    @Counted(name = "createBookmark.count")
+    @Timed(name = "createBookmark.time")
+    public Response createBookmark(@Valid Bookmark bookmark) {
+            bookmark.persist();
+            return Response.status(Response.Status.CREATED).entity(bookmark).build();
     }
 
     @PUT
-    @Path("/{id}")
+    @Path("{id}")
     @Transactional
-    public void update(Bookmark bookmark){
-        Bookmark existing = Bookmark.findById(bookmark.id);
-        existing.url = bookmark.url;
-        existing.description = bookmark.description;
-        existing.title = bookmark.title;
+    @Operation(summary = "Update a bookmark")
+    @Counted(name = "updateBookmark.count")
+    @Timed(name = "updateBookmark.time")
+    public Response updateBookmark(@Valid Bookmark bookmark, @PathParam("id") Long id, @Context UriInfo uriInfo) {
+        Bookmark entity = Bookmark.findById(id);
+        entity.description = bookmark.description;
+        entity.location = bookmark.location;
+        entity.title = bookmark.title;
+        entity.url = bookmark.url;
+        return Response.created(URI.create(uriInfo.getPath())).build();
     }
 
     @DELETE
-    @Path("/{id}")
+    @Path("{id}")
     @Transactional
-    public void delete(@PathParam("id")Long id){
-        Bookmark existing = Bookmark.findById(id);
-        existing.delete();
+    @Operation(summary = "Delete a bookmark")
+    @Counted(name = "deleteBookmark.count")
+    @Timed(name = "deleteBookmark.time")
+    public Response deleteBookmark(@PathParam("id") Long id) {
+        Bookmark bookmark = Bookmark.findById(id);
+        if (bookmark.isPersistent()) {
```

Comment:
Change this to a null check, if bookmark is not null it will always be persistent as we just recover it from the DB